### PR TITLE
[Native/guide] Add information about checking & requesting multiple privileges in single callback

### PR DESCRIPTION
**Pull request by xerrni**
Signed-off-by: Ernest Borowski <e.borowski@partner.samsung.com>

### Change Description ###

Add information about checking & requesting multiple privileges in single callback

 - ACR: ACR-1253
http://suprem.sec.samsung.net/jira/browse/ACR-1253

### CODE REVIEW

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 1]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+    ```
+
 ## Related Information
-- Dependencies
-  - Tizen 2.3 and Higher for Mobile
-  - Tizen 2.3.1 and Higher for Wearable
+- Dependencies:
+  - Tizen 4.0 and Higher for Mobile
+  - Tizen 4.0 and Higher for Wearable
+  - Multiple privileges in single API call:
```
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: ok. thanks

[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: Done

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 2]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+                    break;
+                case PRIVACY_PRIVILEGE_MANAGER_REQUEST_RESULT_DENY_ONCE:
+                    /* Show a message with explanation */
+                    break;
+            }
+        }
+    }
+    ```
+
+## Note
```
Comment (divyk) [inline docs/application/native/guides/security/privacy-related-permissions.md]: Please use the following style for note:

>**Note**
>
>Multiple privileges in single API call are supported from Tizen 5.0.

**Comment by divyk** [comment]
@mijin85cho , I have just done the delta review. Should I review the full doc? Please advice.

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 3]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+   - If the result value is `PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ASK`, the application must request permission from the user with the `ppm_request_permissions()` function, which displays a dialog box. You can pass user data as the forth parameter of the function, as a pointer to an application context instance. When the user makes a decision, a callback defined as the second parameter is invoked.
+
+     The dialog box asking for user permissions is shown only if the `ppm_request_permissions()` function returns without an error.
+
+    ```
+                case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ASK:
+                   askable_privileges[askable_privileges_count++] = privileges[it];
+                   /* Log and handle errors */
+                   break;
+                }
```
Comment (divyk) [inline docs/application/native/guides/security/privacy-related-permissions.md]: Line 185- 194, can be indented in....

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 4]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
 > **Note**  
 > Since the privileges are grouped, the user's decision regarding one privilege applies to the whole group of related privileges. For example, if the user has granted permission to use the `http://tizen.org/privilege/account.read` privilege, permission is automatically granted to the `http://tizen.org/privilege/account.write` privilege also. Be aware that both privileges need to be declared in the application manifest file. If you declare only one of them, the above rule does not apply.
 
+## Requesting Multiple Permissions
+
+This guide describes how to ask and request multiple privileges in single API call. (This API was introduced in Tizen 5.0)
```
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: thanks for comment.
Done

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 5]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+To check whether an application has permission to use a privileges, and to request permissions if required:
+
+1. To check whether an application has permission to use privileges, use the `ppm_check_permissions()` function:
```
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: privilege should be in plural form.
Check whether an application has permission to use privileges, using the `ppm_check_permissions()` function:

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 6]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+   - If the result value is `PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_DENY`, the application is not allowed to perform operations related to the privilege. Any attempt to use such functionality without the user's consent fails. Usually, this means that invoking any API function that involves the privilege results in an error.
+
+     ```
+                case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_DENY:
+                    /* Show a message and terminate the application */
+                    break;
+
+     ```
+
+   - If the result value is `PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ASK`, the application must request permission from the user with the `ppm_request_permissions()` function, which displays a dialog box. You can pass user data as the forth parameter of the function, as a pointer to an application context instance. When the user makes a decision, a callback defined as the second parameter is invoked.
```
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: 1. yes, that`s correct.
This function has four parameters ( comma "," acts as parameter separator)
```c
ppm_request_permissions(const char **privileges, size_t privileges_count,
                            ppm_request_multiple_response_cb callback,
                            void *user_data);
```
The fourth parameter is `void *user_data`.

2. I changed `forth` to `fourth`
thanks for finding this misspell.

File: docs/application/native/guides/security/privacy-related-permissions.md (hunk 2 above)
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: Done

File: docs/application/native/guides/security/privacy-related-permissions.md (hunk 3 above)
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: lines 182 - 184 had wrong indent.
Lines from 185 to 194 are correct.

**Comment by mijin85cho** [comment]
@divyk No, you don't need to review the full docs. Just refer other parts as the reference to get the consistency, and only review the delta part, as you did. : )

**Comment by divyk** [comment]
> @divyk No, you don't need to review the full docs. Just refer other parts as the reference to get the consistency, and only review the delta part, as you did. : )

Ok, thanks, @mijin85cho

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 7]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+                case PRIVACY_PRIVILEGE_MANAGER_REQUEST_RESULT_DENY_ONCE:
+                    /* Show a message with explanation */
+                    break;
+            }
+        }
+    }
+    ```
+
+    > **Note**
+    > Multiple privileges in single API call are supported from Tizen 5.0.
```
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: done.

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 8]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+        int ret = ppm_check_permissions(privileges, sizeof(privileges) / sizeof(privileges[0]), results);
+    ```
+
+    The results of the call is stored in the variable passed as the second parameter (for available results, see the `ppm_check_result_e` enumeration in [mobile](../../api/mobile/latest/group__CAPI__PRIVACY__PRIVILEGE__MANAGER__MODULE.html#ga41f409d8b9d4c27d41b907bdb0975f0c) and [wearable](../../api/wearable/latest/group__CAPI__PRIVACY__PRIVILEGE__MANAGER__MODULE.html#ga41f409d8b9d4c27d41b907bdb0975f0c) applications).
+
+2. React to the permissions check appropriately:
+
+   - If the result value is `PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ALLOW`, the application is allowed to perform operations related to the privilege. For example, the application can enable additional UI elements or functionalities.
+
+    ```
```
[the remark this answers is not in the document]
Comment (divyk) [reply docs/application/native/guides/security/privacy-related-permissions.md]: @mijin85cho, sure, will take care of the layout, too.

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 9]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
 > **Note**  
 > Since the privileges are grouped, the user's decision regarding one privilege applies to the whole group of related privileges. For example, if the user has granted permission to use the `http://tizen.org/privilege/account.read` privilege, permission is automatically granted to the `http://tizen.org/privilege/account.write` privilege also. Be aware that both privileges need to be declared in the application manifest file. If you declare only one of them, the above rule does not apply.
 
+## Requesting Multiple Permissions
+
+This section describes how to check and request multiple privileges in a single API call. This API was introduced in Tizen 5.0.
```
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: done

File: docs/application/native/guides/security/privacy-related-permissions.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: done

File: docs/application/native/guides/security/privacy-related-permissions.md [hunk 10]
```diff
@@ -128,7 +128,105 @@ To check whether an application has permission to use a privilege, and to reques
...
+                    /* Show a message and terminate the application */
+                    break;
+
+     ```
+
+   - If the result value is `PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ASK`, the application must request permission from the user with the `ppm_request_permissions()` function, which displays a dialog box. You can pass user data as the fourth parameter of the function, as a pointer to an application context instance. When the user makes a decision, a callback defined as the second parameter is invoked.
+
+     The dialog box asking for user permissions is shown only if the `ppm_request_permissions()` function returns without an error.
+
+    ```
```
[the remark this answers is not in the document]
Comment (xerrni) [reply docs/application/native/guides/security/privacy-related-permissions.md]: done

**Comment by mijin85cho** [comment]
Thanks for the update.

@divyk Would you please re-approve this PR after checking the updated version? With you approval, I'll merge this PR.

**Comment by mijin85cho** [comment]
Thanks for the review & update.

**Comment by mijin80lee** [comment]
TD Sync Completed.(for Tizen_5.0_prepare)

**Pull request merged**